adi/ad4630.py

```diff
@@ -166,3 +170,70 @@ def calibscale(self):
         def calibscale(self, calibscale):
             """Set calibration scale value."""
             self._set_iio_attr(self.name, "calibscale", False, calibscale, self._ctrl)
+
+
+class adaq42xx(ad4630):
+
+    """ ADAQ4224 is a 24-bit precision SAR ADC data acquisition module """
+
+    def __init__(self, uri="", device_name="adaq4224"):
+
+        context_manager.__init__(self, uri, self._device_name)
+
+        compatible_parts = ["adaq4224", "adaq4216", "adaq4220"]
+
+        if device_name not in compatible_parts:
+            raise Exception(
+                "Not a compatible device: "
+                + str(device_name)
+                + ". Please select from "
+                + str(compatible_parts)
+            )
+        else:
+            self._ctrl = self._ctx.find_device(device_name)
+            self._rxadc = self._ctx.find_device(device_name)
+
+        self.__init_channels__(self)
+
+    class _channel(attribute):
+        """ADAQ42xx differential channel."""
+
+        def __init__(self, ctrl, channel_name):
+            self.name = channel_name
+            self._ctrl = ctrl
+
+        @property
+        def calibbias(self):
+            """Get calibration bias/offset value."""
+            return self._get_iio_attr(self.name, "calibbias", False, self._ctrl)
+
+        @calibbias.setter
+        def calibbias(self, calibbias):
+            """Set calibration bias/offset value."""
+            self._set_iio_attr(
+                self.name, "calibbias", False, int(calibbias), self._ctrl
+            )
+
+        @property
+        def calibscale(self):
+            """Get calibration scale value."""
+            return self._get_iio_attr(self.name, "calibscale", False, self._ctrl)
+
+        @calibscale.setter
+        def calibscale(self, calibscale):
+            """Set calibration scale value."""
+            self._set_iio_attr(self.name, "calibscale", False, calibscale, self._ctrl)
+
+        @property
+        def scale_available(self):
+            """Provides all available scale(gain) settings for the ADAQ42xx channel"""
+            return self._get_iio_attr(self.channel[0].name, "scale_available", False)
+
+        @property
+        def scale(self):
+            """ADAQ42xx channel scale"""
+            return float(self._get_iio_attr_str(self.name, "scale", False))
+
+        @scale.setter
+        def scale(self, value):
+            self._set_iio_attr(self.name, "scale", False, str(Decimal(value).real))
```

Comment:
These are the only new ones. Can you inherit from parent the reused methods to reduce duplication as well. Do you know if these exist in the parent driver but just not implemented in that class?

Reply:
No, the scale_available attribute is not available for ad4630/ad4030 devices.
adaq4224 has a PGA gain which allows the user to scale the input voltage to the ADC.
To support that, adaq4224 has the scale_available attribute and also allows setting the current scale through the scale attribute.
Made child's _channel inherit from parent class so it reuses the other methods.